### PR TITLE
rename z_NmiHandlerSet to z_arm_nmi_set_handler

### LINKED ISSUE #49: zephyr: z_NmiHandlerSet used in port is gone
Zephyr commit https://github.com/zephyrproject-rtos/zephyr/commit/77d2490164cbb5dbdd8dee9c0787204b0006f635 renamed this function.

This is breaking the build at least in ports/zephyr/v2.4/memfault_fault_handler.c:

```
static int prv_install_nmi_handler() {
  extern void z_NmiHandlerSet(void (*pHandler)(void));
  z_NmiHandlerSet(MEMFAULT_EXC_HANDLER_NMI);
  return 0;
}
```

I'm not sure what the right fix is -- perhaps a new ports/zephyr/v3.4 which uses the new name, `z_arm_nmi_set_handler`?

### PR DESCRIPTION
**Pull request by maxd-nordic**
This should fix #49

### CODE REVIEW

File: ports/zephyr/v2.4/memfault_fault_handler.c [hunk 1]
```diff
@@ -28,8 +28,8 @@
 // Since we don't use the arguments we match anything with () to avoid
 // compiler warnings and share the same bootup logic
 static int prv_install_nmi_handler() {
-  extern void z_NmiHandlerSet(void (*pHandler)(void));
-  z_NmiHandlerSet(MEMFAULT_EXC_HANDLER_NMI);
+  extern void z_arm_nmi_set_handler(void (*pHandler)(void));
```
Comment (carlescufi) [inline ports/zephyr/v2.4/memfault_fault_handler.c:31]: Is this the right file?
`ports/zephyr/v2.4/memfault_fault_handler.c`
This is for Zephyr v2.4, and back then the function was still named `z_NmiHandlerSet()`?

Comment (maxd-nordic) [reply ports/zephyr/v2.4/memfault_fault_handler.c:31]: Changing it there fixes the build issue. Maybe there needs to be another version.

Comment (carlescufi) [reply ports/zephyr/v2.4/memfault_fault_handler.c:31]: Right. But this might then break the build against older Zephyr versions?

**Comment by noahp** [comment]
Thanks @carlescufi + @maxd-nordic , this should be solved in `0.43.1`:
https://github.com/memfault/memfault-firmware-sdk/releases/tag/0.43.1

(cc #49 )

**Pull request closed without merging**